packages/source-iotsitewise/src/time-series-data/client/edge/getAggregatedPropertyDataPoints.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  GetAssetPropertyAggregatesCommand,
+  IoTSiteWiseClient,
+  TimeOrdering,
+} from '@aws-sdk/client-iotsitewise';
+import { AssetId, AssetPropertyId } from '../../types';
+import { aggregateToDataPoint } from '../../util/toDataPoint';
+import { RESOLUTION_TO_MS_MAPPING } from '../../util/resolution';
+import { toId } from '../../util/dataStreamId';
+import {
+  parseDuration,
+  OnSuccessCallback,
+  ErrorCallback,
+  RequestInformationAndRange,
+  toSiteWiseAssetProperty,
+} from '@iot-app-kit/core';
+import { isDefined } from '../../../common/predicates';
+import { dataStreamFromSiteWise } from '../../dataStreamFromSiteWise';
+import { SITEWISE_PREVIEW_DATE } from '../../util/timeConstants';
+import { flattenRequestInfoByFetch } from '../../util/flattenRequestInfoByFetch';
+
+const getAggregatedPropertyDataPointsForProperty = ({
+  requestInformation,
+  assetId,
+  propertyId,
+  maxResults,
+  onSuccess,
+  onError,
+  nextToken: prevToken,
+  client,
+}: {
+  requestInformation: RequestInformationAndRange;
+  assetId: AssetId;
+  propertyId: AssetPropertyId;
+  maxResults?: number;
+  onError: ErrorCallback;
+  onSuccess: OnSuccessCallback;
+  client: IoTSiteWiseClient;
+  nextToken?: string;
+}) => {
+  let { start, end } = requestInformation;
+
+  // only 1 of the following options are enabled at this point:
+  // fetchMostRecentBeforeStart, fetchMostRecentBeforeEnd, fetchFromStartToEnd
+  const {
+    aggregationType,
+    fetchMostRecentBeforeStart,
+    fetchMostRecentBeforeEnd,
+    resolution,
+  } = requestInformation;
+
+  if (aggregationType == null) {
+    console.error(
```

Comment:
Not calling `onError` to avoid polluting the other non-aggregate data in the cache